src/api/app/controllers/webui/staging_workflows_controller.rb

```diff
@@ -40,6 +39,26 @@ def show
     @project = @staging_workflow.project
   end
 
+  def destroy
+    @staging_workflow = StagingWorkflow.find_by(id: params[:id])
+    authorize @staging_workflow
+    @project = @staging_workflow.project
+
+    # attached staging projects get nullified by default but we want to
+    # allow to destroy manually by setting a checkbox
+    @staging_workflow.staging_projects.where(id: params[:staging_project_ids]).destroy_all
+
+    respond_to do |format|
+      if @staging_workflow.destroy
+        format.js { flash[:success] = "Staging Workflow for #{@project.name} was successfully deleted." }
```

Comment:
You can set the flash messages outside of the `format.js` blocks. And then you won't need the whole `respond_to` part.

Reply:
done :)